src/app/frontend/deploy/deployfromsettings.html

```diff
@@ -17,17 +17,19 @@
 <kd-help-section>
   <md-input-container class="md-block" md-is-error="ctrl.isNameError()">
     <label>App name</label>
-    <input ng-model="ctrl.name" name="name" namespace="ctrl.namespace" required kd-unique-name
-        ng-model-options="{ updateOn: 'default blur', debounce: { 'default': 500, 'blur': 0 } }">
-    <md-progress-linear class="kd-deploy-form-progress" md-mode="indeterminate"
-        ng-show="ctrl.form.name.$pending">
-    </md-progress-linear>
-    <ng-messages for="ctrl.form.name.$error" role="alert" multiple>
-      <ng-message when="required">Application name is required.</ng-message>
-      <ng-message when="uniqueName">
-        Application with this name already exists within namespace <i>{{ctrl.namespace}}</i>.
-      </ng-message>
-    </ng-messages>
+    <input ng-model="ctrl.name"
+           name="name"
+           namespace="ctrl.namespace"
+           required
+           kd-unique-name
+           ng-pattern="/^[a-z]+[a-z\-]*[a-z]*$/">
+
+    <div ng-messages="ctrl.form.name.$error" ng-if="ctrl.form.name.$touched">
```

Comment:
This addition still does not cause validation until the first blur. The user only starts seeing any validation after the first time he/she clicks away. After this however is where there is a difference: without this, if there are two errors in a row, the error mesage disappears. The input area remains red, but with no help text. For example, I type " %34_invalidName ", the pattern error shows. I back-space to remove all the characters one by one. The last character that I remove was "%" - (still an error character, so I haven't trggered a valid state yet). When I delete this last invalid character, the input component goes straight from one error state to the next ('required' is now not met). However the error message disappears altogether. 
this `ng-if="ctrl.form.name.$touched"` fixes that.